command_line_assistant/utils/files.py

```diff
@@ -93,3 +100,94 @@ def write_file(contents: Union[str, bytes], path: Path, mode: int = 0o600) -> No
             path,
             str(e),
         )
+
+
+class NamedFileLock:
+    """File-based lock mechanism for general locking purposes.
+
+    This class provides a way to track state across processes
+    using a lock file. The lock file contains the PID of the capturing process.
+
+    Usage:
+        with NamedFileLock() as lock:
+            # Execute your operations
+        # Lock is automatically released
+    """
+
+    def __init__(self, name: str):
+        """Initialize the named file lock mechanism."""
+        self._pid = os.getpid()
+        self._name = name
+        self._lock_file = Path(get_xdg_state_path(), name + ".lock")
```

Comment:
Does it make sense to do any string validation on `name` before we use it as a file name in a path? Like, should we replace whitespace with underscores, or raise an exception if the string doesn't conform to certain criteria?

Reply:
This is meant to be used by developers only. I don't expected anyone else creating locking objects outside the code, and such things we could prevent/catch during reviews. We could throw an exception in case the name is not a str object or conform to a minimal criterion, but I would not be too strict to that